internal/vault/credentialproviders/jwt.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package credentialproviders
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	secretsv1alpha1 "github.com/hashicorp/vault-secrets-operator/api/v1alpha1"
+	authv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type JwtCredentialProvider struct {
+	authObj           *secretsv1alpha1.VaultAuth
+	providerNamespace string
+	uid               types.UID
+}
+
+func (l *JwtCredentialProvider) GetNamespace() string {
+	return l.providerNamespace
+}
+
+func (l *JwtCredentialProvider) GetUID() types.UID {
+	return l.uid
+}
+
+func (l *JwtCredentialProvider) Init(ctx context.Context, client ctrlclient.Client, authObj *secretsv1alpha1.VaultAuth, providerNamespace string) error {
+	l.authObj = authObj
+	l.providerNamespace = providerNamespace
+
+	if l.authObj.Spec.Jwt.ServiceAccount != "" {
+		sa, err := l.getServiceAccount(ctx, client)
+		if err != nil {
+			return err
+		}
+		l.uid = sa.UID
+	} else if l.authObj.Spec.Jwt.Token != "" {
+		l.uid = types.UID(uuid.New().String())
+	} else {
+		return fmt.Errorf("either serviceAccount or jwt is required in VaultAuth Custom Resource to" +
+			" retrieve credentials to authenticate to Vault's jwt authentication backend")
+	}
+
+	return nil
+}
+
+func (l *JwtCredentialProvider) getServiceAccount(ctx context.Context, client ctrlclient.Client) (*corev1.ServiceAccount, error) {
+	key := ctrlclient.ObjectKey{
+		Namespace: l.providerNamespace,
+		Name:      l.authObj.Spec.Kubernetes.ServiceAccount,
+	}
+	sa := &corev1.ServiceAccount{}
+	if err := client.Get(ctx, key, sa); err != nil {
+		return nil, err
+	}
+	return sa, nil
+}
+
+func (l *JwtCredentialProvider) GetCreds(ctx context.Context, client ctrlclient.Client) (map[string]interface{}, error) {
```

Comment:
Will this support grabbing a jwt token from a k8s secret as well?

Reply:
Yes, it will.